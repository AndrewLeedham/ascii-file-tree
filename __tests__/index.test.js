const index = require('../index');
const fg = require('fast-glob');
const mockGlobs = require('./__fixtures__/globs.json');

const globSpy = jest.spyOn(fg, 'sync');

beforeEach(() => {
  globSpy.mockImplementation(() => mockGlobs.control);
});

describe('generate', () => {
  test('control', () => {
    expect(index.generate()).toMatchSnapshot();
  });

  test('nesting', () => {
    globSpy.mockImplementation(() => mockGlobs.nesting);
    expect(index.generate()).toMatchSnapshot();
  });

  test('deep nesting', () => {
    globSpy.mockImplementation(() => mockGlobs.deepNesting);
    expect(index.generate()).toMatchSnapshot();
  });

  test('branching', () => {
    globSpy.mockImplementation(() => mockGlobs.branching);
    expect(index.generate()).toMatchSnapshot();
  });
});
