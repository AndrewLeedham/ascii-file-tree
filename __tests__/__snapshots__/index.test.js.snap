// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate branching 1`] = `
" .
├─ .gitignore
├─ package.json
└─ src
   └─ one
      ├─ one.js
      ├─ two
      │  └─ three
      │     └─ three.js
      └─ branch
         └─ two
            └─ two.js"
`;

exports[`generate control 1`] = `
" .
├─ .gitignore
├─ package.json
└─ src
   └─ index.js"
`;

exports[`generate deep nesting 1`] = `
" .
├─ .gitignore
├─ package.json
└─ src
   └─ really
      └─ really
         └─ deep
            └─ nesting
               ├─ other.js
               └─ that
                  └─ could
                     └─ confuse
                        └─ things
                           └─ index.js"
`;

exports[`generate nesting 1`] = `
" .
├─ .gitignore
├─ package.json
└─ src
   ├─ index.js
   └─ __tests__
      └─ index.test.js"
`;
